shared/gh/js/views/gh.delete-module.js

```diff
@@ -0,0 +1,350 @@
+                orgUnitAPI.deleteOrgUnit(moduleId, function(err) {
+                    if (err) {
+                        return gh.utils.notification('Could not delete the module', constants.messaging.default.error, 'error');
+                    }
+
+                    // Hide the modal
+                    $('#gh-delete-module-modal').modal('hide');
+
+                    // Show a success notification after the module and all its series have been deleted
+                    gh.utils.notification('The module has been removed', null, 'success');
+
+                    // Remove the module from the navigation
+                    $('.list-group-item[data-id="' + moduleId + '"]').remove();
+
+                    // Remove the module and series from the state
+                    gh.utils.removeFromState(['series', 'module']);
+                });
+            });
+        });
+
+        // Avoid default form submit behaviour
+        return false;
+    };
+
+
+    ////////////////////
+    // INITIALISATION //
+    ////////////////////
+
+    /**
+     * Render and show the 'delete module' modal dialog
+     *
+     * @private
+     */
+    var setUpDeleteModuleModal = function() {
+        // Cache the module ID
+        moduleId = parseInt($(this).attr('data-id'), 10);
+
+        // Render the modal
+        gh.utils.renderTemplate($('#gh-delete-module-modal-template'), {'data': null}, $('#gh-delete-module-modal-container'));
+
+        // Show the modal
+        $('#gh-delete-module-modal').modal();
+    };
+
+    /**
+     * Create a list of organisational units from which series are borrowed from
+     *
+     * @param  {Object}      orgunit     The organisational unit (module) to delete
+     * @param  {Function}    callback    Standard callback function
+     * @private
+     */
+    var retrieveBorrowedOrgUnits = function(orgunit, callback) {
+        borrowedElsewhere = [];
+        _.each(orgunit.Series, function(series) {
+            _.each(series.OrgUnits, function(orgunit, index) {
+                // If the first orgunit in the list is the current module,
+                // any next modules have borrowed this series and should be added to the list
+                if (index !== 0 && series.OrgUnits[0].id === moduleId) {
+                    orgunit.seriesId = series.id;
+                    borrowedElsewhere.push(orgunit);
+                }
+            });
+        });
+
+        // Make sure each entry is unique
+        borrowedElsewhere = _.uniq(borrowedElsewhere, function(item) {
+            return item.id;
+        });
+
+        var todo = borrowedElsewhere.length;
+        var done = 0;
+
+        /**
+         * Get the parent organisational units for modules which borrow a series from the module to delete
+         *
+         * @param  {Object}      module       The module to get the parent organisational units for
+         * @param  {Function}    _callback    Standard callback function
```

Comment:
If these requests fail there's no point in even executing the callback. Instead, I'll just return a notification message that indicates a problem on our side (default since oszkar's review of the notification messages)